pkg/crosscluster/logical/replication_statements.go

```diff
@@ -196,6 +224,184 @@ func newDeleteStatement(
 	return toParsedStatement(delete)
 }
 
+// newBulkSelectStatement returns a statement that can be used to query multiple rows
+// by primary key in a single operation. It uses a CTE with UNNEST to handle a variable
+// number of primary keys provided as array parameters.
+//
+// The statement will have one parameter that is the index of each input row and
+// one parameter for each primary key column, where each parameter is an array
+// of values for that column. The columns are expected in column ID order, not
+// primary key order.
+//
+// For example, given a table with primary key columns (id, secondary_id) and additional
+// columns (value1, value2), the generated statement would be equivalent to:
+//
+//	WITH key_list AS (
+//	  SELECT unnest($1::INT[]) as index, unnest($2::INT[]) as key1, unnest($3::INT[]) as key2
+//	)
+//	SELECT key_list.index,
+//	       replication_target.crdb_internal_origin_timestamp,
+//	       replication_target.crdb_internal_mvcc_timestamp,
+//	       replication_target.id, replication_target.secondary_id,
+//	       replication_target.value1, replication_target.value2
+//	FROM table_id AS replication_target
+//	INNER JOIN key_list ON replication_target.id = key_list.key1
+//	                    AND replication_target.secondary_id = key_list.key2
```

Comment:
Should this be `SELECT ... FOR UPDATE OF replication_target`? (In other words, are we going to issue the mutations in the same transaction as this read?)

Reply:
This is something I'm thinking of tweaking when measuring the performance of the select. For now the select is in its own transaction, but I intended to test if a locking statement in the same transaction performs better.